### DISCUSSION TITLE
Automate build

### DISCUSSION THREAD

**Discussion by BruceWestbury** [executablebooks/meta#165] [Q&A]
Can I improve my workflow? This consists in:

edit, ..., edit, build, refresh page in browser, edit, ... (repeat)

I am using VSCode as my editor. Can I set this so the build and refresh happen automatically
every time I save a file? I use VSCode for other editing e.g. python code so I only
want this to happen when I am working on a book.

**Comment by choldgraf** [comment]
Right now what I do is use this useful VSCode plugin:

https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer

e.g.:

![](https://github.com/ritwickdey/vscode-live-server/raw/master/images/Screenshot/vscode-live-server-animated-demo.gif)

then you just navigate your server to the `_build/html` folder and it'll open a browser window that auto-updates when you re-build your book 👍 

Longer-term, I think it'd be great if Jupyter Book had something like `jupyter-book build mybook/ --watch` that would run the server for you
